src/test/java/org/broadinstitute/hellbender/tools/sv/cluster/SVCollapserTest.java

```diff
@@ -0,0 +1,783 @@
+        Assert.assertEquals((int) resultMaxMin.getKey(), expectedMaxMin[0]);
+        Assert.assertEquals((int) resultMaxMin.getValue(), expectedMaxMin[1]);
+
+        final Pair<Integer, Integer> resultMean = collapserMean.collapseInterval(records);
+        Assert.assertEquals((int) resultMean.getKey(), expectedMean[0]);
+        Assert.assertEquals((int) resultMean.getValue(), expectedMean[1]);
+    }
+
+    @DataProvider(name = "collapseTypesTestData")
+    public Object[][] collapseTypesTestData() {
+        return new Object[][]{
+                {Collections.singletonList(StructuralVariantType.DEL), StructuralVariantType.DEL},
+                {Collections.singletonList(StructuralVariantType.DUP), StructuralVariantType.DUP},
+                {Collections.singletonList(StructuralVariantType.INS), StructuralVariantType.INS},
+                {Collections.singletonList(StructuralVariantType.INV), StructuralVariantType.INV},
+                {Collections.singletonList(StructuralVariantType.BND), StructuralVariantType.BND},
+                {Collections.singletonList(StructuralVariantType.CNV), StructuralVariantType.CNV},
+                {Lists.newArrayList(StructuralVariantType.DEL, StructuralVariantType.DUP), StructuralVariantType.CNV},
+                {Lists.newArrayList(StructuralVariantType.DEL, StructuralVariantType.DEL), StructuralVariantType.DEL},
+                {Lists.newArrayList(StructuralVariantType.DUP, StructuralVariantType.DUP), StructuralVariantType.DUP},
+                {Lists.newArrayList(StructuralVariantType.INS, StructuralVariantType.INS), StructuralVariantType.INS},
+                {Lists.newArrayList(StructuralVariantType.INV, StructuralVariantType.INV), StructuralVariantType.INV},
+                {Lists.newArrayList(StructuralVariantType.BND, StructuralVariantType.BND), StructuralVariantType.BND},
+                {Lists.newArrayList(StructuralVariantType.CNV, StructuralVariantType.CNV), StructuralVariantType.CNV},
+                {Lists.newArrayList(StructuralVariantType.DEL, StructuralVariantType.DUP, StructuralVariantType.CNV), StructuralVariantType.CNV}
+        };
+    }
+
+    @Test(dataProvider= "collapseTypesTestData")
+    public void collapseTypesTest(final List<StructuralVariantType> types, final StructuralVariantType expectedResult) {
+        final List<SVCallRecord> records = types.stream().map(t -> SVTestUtils.newCallRecordWithIntervalAndType(1, 100, t)).collect(Collectors.toList());
+        Assert.assertEquals(collapser.collapseTypes(records), expectedResult);
+    }
+
+    @DataProvider(name = "collapseAlgorithmsTestData")
+    public Object[][] collapseAlgorithmsTestData() {
+        return new Object[][]{
+                {Collections.singletonList(Collections.singletonList(GATKSVVCFConstants.DEPTH_ALGORITHM)), Collections.singletonList(GATKSVVCFConstants.DEPTH_ALGORITHM)},
+                {Collections.singletonList(Collections.singletonList("pesr")), Collections.singletonList("pesr")},
```

Comment:
Are there cases where events can be supported by depth and pesr evidence?

Reply:
Yes, added those cases now